### PR TITLE
added _from and _to indexes to GetAllUsersData and GetAllMyTokens functions

### LINKED ISSUE #107: ID 10: Medium - Breakable getters
Present at: Delay View Contract

Description: Getters such as GetAllUsers data and GetAllMyTokens are returning full dynamic arrays. Since there's a computational limit to the on-chain evm functions, such functions can become unusable once arrays reach certain sizes.

Recommendation: Add arguments 'from' and 'to' to the functions, enabling selection of data sets to return (instead of returning whole array at all times, you will be able to select from which to which array index you want to return the data).

### PR DESCRIPTION
**Pull request by YouStillAlive**
closes #107

### CODE REVIEW

File: contracts/DelayView.sol [hunk 1]
```diff
@@ -5,24 +5,37 @@ import "./DelayManageable.sol";
...
+    function GetAllUsersDataRange(
+        address _token,
+        uint256 _from,
+        uint256 _to
+    ) external view returns (address[] memory _users, Vault[] memory _vaults) {
+        require(_from <= _to, "_from index can't be greater than _to");
+        require(
+            _from < Users[_token].length || _to < Users[_token].length,
+            "index out of range"
+        );
```
Comment (ashwinarora) [inline contracts/DelayView.sol:17]: need AND here

File: contracts/DelayView.sol [hunk 2]
```diff
@@ -5,24 +5,37 @@ import "./DelayManageable.sol";
...
+    function GetAllMyTokensRange(
+        address _user,
+        uint256 _from,
+        uint256 _to
+    ) external view returns (address[] memory _tokens) {
+        require(_from <= _to, "_from index can't be greater than _to");
+        require(
+            _from < MyTokens[_user].length || _to < MyTokens[_user].length,
+            "index out of range"
+        );
```
Comment (ashwinarora) [inline contracts/DelayView.sol:34]: need AND here

**Review by ashwinarora — CHANGES_REQUESTED** [review]

File: contracts/DelayView.sol [hunk 3]
```diff
@@ -5,24 +5,37 @@ import "./DelayManageable.sol";
 
 /// @title DelayView - getter view functions
 contract DelayView is DelayManageable {
-    function GetAllUsersData(address _token)
-        external
-        view
-        returns (address[] memory, Vault[] memory _vaults)
-    {
-        _vaults = new Vault[](Users[_token].length);
-        for (uint256 i = 0; i < Users[_token].length; i++) {
```
Comment (Lomet) [inline contracts/DelayView.sol:14]: do we have a getter for 
`Users[_token].length`

File: contracts/DelayView.sol [hunk 4]
```diff
@@ -5,24 +5,37 @@ import "./DelayManageable.sol";
...
+    function GetAllUsersDataRange(
```
Comment (ashwinarora) [inline contracts/DelayView.sol:8]: rename to `GetUsersDataByRange`

File: contracts/DelayView.sol [hunk 5]
```diff
@@ -5,24 +5,37 @@ import "./DelayManageable.sol";
...
-        return (Users[_token], _vaults);
     }
 
-    function GetAllMyTokens(address _user)
-        external
-        view
-        returns (address[] memory)
-    {
-        return MyTokens[_user];
+    function GetAllMyTokensRange(
```
Comment (ashwinarora) [inline contracts/DelayView.sol:25]: rename to `GetMyTokensByRange`

**Review by ashwinarora — CHANGES_REQUESTED** [review]

File: contracts/DelayView.sol (hunk 3 above)
Comment (ashwinarora) [reply contracts/DelayView.sol:14]: not yet, we'll need one

**Review by Lomet — APPROVED** [review]

File: contracts/DelayView.sol [hunk 6]
```diff
@@ -5,24 +5,44 @@ import "./DelayManageable.sol";
...
+        _tokens = new address[](_to - _from + 1);
+        for (uint256 i = _from; i <= _to; i++) {
+            _tokens[i] = MyTokens[_user][i];
+        }
+    }
+
+    function GetUsersLengthByToken(
+        address _token
+    ) external view returns (uint256) {
+        return TokenToUsers[_token].length;
```
Comment (ashwinarora) [inline contracts/DelayView.sol:45]: Think we need another view function `GetMyTokensLengthByUser`

**Review by ashwinarora — CHANGES_REQUESTED** [review]

**Review by Lomet — APPROVED** [review]

**Pull request merged**